### ISSUE TITLE
Add functions for all query options

### ISSUE THREAD

**Issue by keathley** [keathley/wallaby#404]
Based on #379 we want to create functions that can be used to augment queries. We should start with all of our existing options. These functions should take a query and return a query so that they are composable with other query functions.

The initial functions we should create are:

* `count(query, term())`
* `visible(query, bool)`
* `selected(query, bool())`
* `text(query, String.t())`
* `at(query, pos_integer())`

Originally we had said we were going to use `hidden` instead of `visible` but the option we use is `visible` so its better to keep those consistent.

**Comment by germsvel** [comment]
I started working on an implementation for this, and I ran into a possible issue. The `text(query, String.t())` function might cause confusion with [`text(selector, opts \\ [])` ](https://hexdocs.pm/wallaby/Wallaby.Query.html#text/2) since both of them are `Query.text/2` functions. 

We can pattern match on `%Query{}` in the implementation to know which of those is being called, but I thought I'd ask here to see if that's okay, or if you think we should change the name of the function to something like `put_text(query, String.t())`. Changing that name might mean we might want to change the name of all the other functions, so not sure that's a great way to go.
